tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/LogsInjectionTestBase.cs

```diff
@@ -120,10 +119,64 @@ public void ValidateLogCorrelation(
                 additionalInjectedLogFilter ??= (_) => true;
                 var tracedLogs = logs.Where(log => !log.Contains(ExcludeMessagePrefix)).Where(additionalInjectedLogFilter).ToList();
 
+                tracedLogs.Should().NotBeNullOrEmpty();
+
                 // Ensure that all spans are represented (when correlated) or no spans are represented (when not correlated) in the traced logs
                 if (tracedLogs.Any())
                 {
-                    var traceIds = spans.Select(x => x.TraceId.ToString()).Distinct().ToList();
+                    List<string> traceIds = new List<string>();
+                    if (use128Bits)
+                    {
+                        // dumb but simple way of reassembling 128 bit trace ids
+                        // we first collect all spans that have a propagated trace id upper
+                        // we build a dictionary of {upper: lower} trace ids
+                        // then we go back through the spans and reassemble the 128 bit trace id
+                        // there isn't a clean way of getting the 128-bit id from the MockSpan
+                        // This could be converted to a LINQ query but...
+                        var lowerUpper = new Dictionary<string, string>();
+                        foreach (var span in spans)
+                        {
+                            var lower = span.TraceId.ToString("x16");
+                            var upper = span.GetTag(Tags.Propagated.TraceIdUpper);
+                            if (string.IsNullOrEmpty(upper))
+                            {
+                                continue;
+                            }
+
+                            if (lowerUpper.ContainsKey(lower))
+                            {
+                                continue;
+                            }
+                            else
+                            {
+                                lowerUpper.Add(lower, upper);
```

Comment:
This assumes there's only ever one span in the trace, doesn't it? 🤔 Is this safer, incase there are two spans in the trace?

```suggestion
                                lowerUpper[lower] = upper;
```

EDIT: I take it back, I don't really know what any of these loops are doing 😅

Reply:
> I don't really know what any of these loops are doing

When serialized, all spans contain the lower 64 bits of a 128-bit trace id, but only one span in each trace chunk contains the higher 64 bits. The loops are trying to reconstruct all the trace ids in their full 128-bit glory from the deserialized span data.

Reply:
I've reverted to just getting the spans that have `_dd.t.tid` and combining to get the 128-bit ID for them and regexing on those.